### ISSUE TITLE
Angular Ivy can't recompile

### ISSUE THREAD

**Issue by Ploppy3** [angular/angular#31512]
# 🐞 bug report

### Affected Package

 Unknown

### Is this a regression?

(Please read description first, not sure why the fields are in this order)
I also tested 8.2.0-next-0, for this one my app was recompiling but no changes could be seen.

### Description

 Testing Ivy (8.2.0-next-1) to see the benefits of the faster recompilation during development, my app loads fine at first, but then after it recompiles on small changes (example: modify html text), it brakes returning errors related to the content of the page ; from on page to an other, the error can be different.

## 🔥 Exception or Error
On one page:
![image](https://user-images.githubusercontent.com/7990765/61050694-e23e0900-a3e7-11e9-8a5b-66bc0abec507.png)

On another page:
![image](https://user-images.githubusercontent.com/7990765/61051801-419d1880-a3ea-11e9-9cde-bc9f757fed78.png)

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 8.2.0-next.0
Node: 12.3.1
OS: win32 x64
Angular: 8.2.0-next.1
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
... service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.802.0-next.0
@angular-devkit/build-angular     0.802.0-next.0
@angular-devkit/build-optimizer   0.802.0-next.0
@angular-devkit/build-webpack     0.802.0-next.0
@angular-devkit/core              8.2.0-next.0
@angular-devkit/schematics        8.2.0-next.0
@angular/cdk                      8.0.2
@angular/cli                      8.2.0-next.0
@angular/language-service         8.0.3
@angular/material                 8.0.2
@angular/pwa                      0.12.4
@ngtools/webpack                  8.2.0-next.0
@schematics/angular               7.2.4
@schematics/update                0.802.0-next.0
rxjs                              6.5.2
typescript                        3.4.5
webpack                           4.35.3
</code></pre>

<pre><code>
{
  "name": "",
  "version": "1.9.3",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~8.2.0-next.1",
    "@angular/cdk": "~8.0.1",
    "@angular/common": "~8.2.0-next.1",
    "@angular/compiler": "~8.2.0-next.1",
    "@angular/core": "~8.2.0-next.1",
    "@angular/forms": "~8.2.0-next.1",
    "@angular/material": "^8.0.1",
    "@angular/platform-browser": "~8.2.0-next.1",
    "@angular/platform-browser-dynamic": "~8.2.0-next.1",
    "@angular/pwa": "0.12",
    "@angular/router": "~8.2.0-next.1",
    "@angular/service-worker": "~8.2.0-next.1",
    "@auth0/angular-jwt": "^2.1.0",
    "@types/randomcolor": "^0.5.2",
    "chart.js": "^2.8.0",
    "core-js": "^2.5.4",
    "hammerjs": "^2.0.8",
    "moment": "^2.24.0",
    "randomcolor": "^0.5.4",
    "rxjs": "~6.5.2",
    "tslib": "^1.9.0",
    "zone.js": "~0.9.1"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.802.0-next.0",
    "@angular/cli": "~8.2.0-next.0",
    "@angular/compiler-cli": "~8.2.0-next.1",
    "@angular/language-service": "8.2.0-next.1",
    "@types/jasmine": "~2.8.8",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "codelyzer": "~4.5.0",
    "jasmine-core": "~2.99.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~1.1.2",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.4.0",
    "ts-node": "~7.0.0",
    "tslint": "~5.11.0",
    "typescript": "~3.4.5"
  }
}
</code></pre>

**Anything else relevant?**

Note that stopping the server and restarting with `ng serve` will rebuild the app properly without any error.

**Comment by magneticflux-** [comment]
I have encountered this as well. My `package.json`:
```json
{
  "name": "test",
  "version": "0.1.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e",
    "postinstall": "ivy-ngcc"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~8.2.0-next.1",
    "@angular/cdk": "~8.0.2",
    "@angular/common": "~8.2.0-next.1",
    "@angular/compiler": "~8.2.0-next.1",
    "@angular/core": "~8.2.0-next.1",
    "@angular/forms": "~8.2.0-next.1",
    "@angular/material": "^8.0.2",
    "@angular/platform-browser": "~8.2.0-next.1",
    "@angular/platform-browser-dynamic": "~8.2.0-next.1",
    "@angular/router": "~8.2.0-next.1",
    "angular4-hal-redux": "~8.0.0-alpha.0",
    "hammerjs": "^2.0.8",
    "ngx-infinite-scroll": "^7.2.0",
    "rxjs": "~6.5.2",
    "tslib": "^1.10.0",
    "typeface-roboto": "0.0.54",
    "zone.js": "~0.9.1"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "~0.802.0-next.0",
    "@angular/cli": "~8.2.0-next.0",
    "@angular/compiler-cli": "~8.2.0-next.1",
    "@angular/language-service": "~8.2.0-next.1",
    "@types/node": "~12.6.2",
    "@types/jasmine": "~3.3.13",
    "@types/jasminewd2": "~2.0.6",
    "codelyzer": "^5.0.0",
    "jasmine-core": "~3.4.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.2.0",
    "karma-chrome-launcher": "~3.0.0",
    "karma-coverage-istanbul-reporter": "~2.0.5",
    "karma-jasmine": "~2.0.1",
    "karma-jasmine-html-reporter": "^1.4.2",
    "protractor": "~5.4.2",
    "ts-node": "~8.3.0",
    "tslint": "~5.18.0",
    "typescript": "~3.6.0-dev.20190713"
  }
}
```

It works correctly using Angular `8.1`, but then I encounter #31160 instead so I'm forced to use Angular `8.0` for everything to work right again.

**Comment by gkalpak** [comment]
Mentioning likely related issues for context: #31560, #31561

**Comment by alxhub** [comment]
@Ploppy3 @magneticflux- we could use a simple reproduction repro for this case. Thanks!

**Comment by magneticflux-** [comment]
@alxhub Here's a sample that exhibits the issue! https://github.com/magneticflux-/angular-issue31512-repro

**Comment by Ploppy3** [comment]
Not fixed with 8.2.0-next.2

**Comment by alxhub** [comment]
Closing as we're tracking this in the root cause issue #31654
